### ISSUE TITLE
Nested requires which bring in aliases used with keywords cause issues

### ISSUE THREAD

**Issue by jeaye** [jonase/kibit#202]
I've read through #14 and I see that a PR was merged for it here: #198 From what I see, this isn't in the latest stable release, so I've been working on the latest beta `[lein-kibit "0.1.6-beta2"]`. Here is the most minimal reproduction case:

```text
┌─(✓)[jeaye@oryx]─[~/projects/tmp][22:11:46]
└──╼ lein new app kibit-test
Generating a project called kibit-test based on the 'app' template.
┌─(✓)[jeaye@oryx]─[~/projects/tmp][22:11:55]
└──╼ cd kibit-test/
┌─(✓)[jeaye@oryx]─[~/projects/tmp/kibit-test][22:11:57]
└──╼ vim src/kibit_test/core.clj 
┌─(✓)[jeaye@oryx]─[~/projects/tmp/kibit-test][22:12:32]
└──╼ cat src/kibit_test/core.clj 
(ns kibit-test.core
  (:gen-class)
  (:require [clojure
             [string :as string]]))

(defn -main
  [& args]
  (println ::string/uh-oh))
┌─(✓)[jeaye@oryx]─[~/projects/tmp/kibit-test][22:12:37]
└──╼ lein run
:clojure.string/uh-oh
┌─(✓)[jeaye@oryx]─[~/projects/tmp/kibit-test][22:12:48]
└──╼ lein kibit
Check failed -- skipping rest of file
[line 8, col 26] Invalid keyword: ::string/uh-oh.
```

**Comment by danielcompton** [comment]
Thanks for reporting this, I think the reader probably needs to be extended to understand nested requires.

**Comment by arrdem** [comment]
Argh we wrote our own `require` parser at some point didn't we

**Comment by danielcompton** [comment]
Yeah, https://github.com/jonase/kibit/blob/master/kibit/src/kibit/check/reader.clj.

**Comment by jeaye** [comment]
Happy new year! Any update on the priority of this one?

**Comment by arrdem** [comment]
Still no priority on this, but it happens that I wrote a `ns` parser for another project recently so we should be able to get a fix for this out whenever I can find an hour to get that code shuffled around and packaged for use here.

**Comment by jeaye** [comment]
Good to know. Thanks for the update.

**Comment by tomjkidd** [comment]
Another Happy New Year! Will 2019 see some love for this?

**Comment by arrdem** [comment]
Probably not. You gonna send us a patch @tomjkidd?

**Comment by tomjkidd** [comment]
I'm willing to look if it isn't duplicating the work of others!

**Comment by tomjkidd** [comment]
Took a stab at it, please refer to https://github.com/jonase/kibit/pull/226

**Comment by arrdem** [comment]
@tomjkidd Thanks for the PR! On a quick pass it looks good, I'll try to give it more time later today.

**Comment by tomjkidd** [comment]
@arrdem I imagine y'all are pretty busy, so is there any way to help this along?

**Comment by arrdem** [comment]
@tomjkidd Thanks for the reminder, I read through this ~two weeks ago and it looked mostly good but I have some concerns about the vagaries of the require incantation I need to poke at. Will look now.

**Comment by arrdem** [comment]
```
kibit.test.check-reader> (reader/read-file (clojure.lang.LineNumberingPushbackReader. (java.io.StringReader. "(ns user (:require clojure.string)) ::clojure.string/foo")) 'user)
((ns user (:require clojure.string)) :clojure.string/foo)
kibit.test.check-reader> (reader/read-file (clojure.lang.LineNumberingPushbackReader. (java.io.StringReader. "(ns user (:require clojure.string)) ::clojure.string/foo")) 'user)
((ns user (:require clojure.string)) :clojure.string/foo)
kibit.test.check-reader> 
```

Okay. This was the behavior I was concerned could mess up - looks good!

**Comment by arrdem** [comment]
I think we can close this out if/when @danielcompton can be persuaded to cut a release.

**Comment by tomjkidd** [comment]
@arrdem Thanks for taking a look!

**Comment by danielcompton** [comment]
Fixed with the release of 0.1.7: https://github.com/jonase/kibit/releases/tag/0.1.7.
